static/js/util/validation.js

```diff
@@ -10,12 +10,15 @@ let handleNestedValidation = (profile, keys, nestedKey) => {
 };
 
 let checkFieldPresence = (profile, requiredFields, messages) => {
-  return Object.assign({}, ...requiredFields.map( keySet => {
+  let errors = {};
+
+  for (let keySet of requiredFields) {
     let val = _.get(profile, keySet);
     if (_.isUndefined(val) || _.isNull(val) || val === "" ) {
-      return _.set({}, keySet,  `${messages[keySet.slice(-1)[0]]} is required`);
+      _.set(errors, keySet,  `${messages[keySet.slice(-1)[0]]} is required`);
     }
-  }));
+  }
+  return errors;
```

Comment:
Ah I see, before I had written this:

```js
Object.assign({}, ...requiredFields.map (keySet => { ... });
```

where the arrow function returns something that should look like this:

```js
{'work_history': [
  {'fieldName': 'Field Name is required'}
]}
```

so we're splatting a bunch of these objects together, but whichever one ends up being last sets the `work_history` key on the object we end up returning to just the one error. bleh, this is what happens when you get too fancy...